docker/build.py

```diff
@@ -1,85 +1,172 @@
-for framework in frameworks:
-    dockerfile = script_dir / f"Dockerfile-{framework}"
-    if not dockerfile.exists():
-        print("make sure you run this script from the root of the armory repo")
-        raise ValueError(f"Dockerfile not found: {dockerfile}")
-
-    cmd = [
-        "docker",
-        "build",
-        "--file",
-        str(dockerfile),
-        "--tag",
-        f"twosixarmory/{framework}:{armory.__version__}",
-        "--build-arg",
-        f"base_image_tag={args.base_tag}",
-        "--build-arg",
-        f"armory_version={armory.__version__}",
-        "--force-rm",
+root_dir   = script_dir.parent
+
+armory_frameworks  = ["pytorch", "pytorch-deepspeech", "tf2"]
+container_platform = "docker" if shutil.which("docker") else "podman"
+
+
+def cli_parser(argv=sys.argv[1:]):
+    parser    = argparse.ArgumentParser("Armory Container Build Script")
+    arguments = (
+        (("-f", "--framework"), dict(
+            choices  = armory_frameworks + ["all"],
+            help     = "Framework to build",
+            required = True,
+        )),
+        (("-b", "--base-tag"), dict(
+            help     = "Version tag for twosixarmory/armory-base",
+            default  = "latest",
+            required = False,
+        )),
+        (("-nc", "--no-cache"), dict(
+            action = "store_true",
+            help   = "Do not use docker cache",
+        )),
+        (("-np", "--no-pull"), dict(
+            action = "store_true",
+            help   = "Do not pull latest base",
+        )),
+        (("-n", "--dry-run"), dict(
+            action = "store_true",
+            help   = "Do not build, only print commands",
+        )),
+        (("-v", "--verbose"), dict(
+            action = "store_true",
+            help   = "Print verbose output",
+        )),
+        (("-p", "--platform"), dict(
+            choices  = ["docker", "podman"],
+            help     ="Print verbose output",
+            default  = container_platform,
+            required = False,
+        )),
+    )
+    for args, kwargs in arguments:
+        parser.add_argument(*args, **kwargs)
+    parser.set_defaults(func=init)
+    return parser.parse_args(argv)
+
+
+def move_file(src, dest, name=False):
+    if dest.is_dir():
+        dest_abs  = dest.absolute()
+        dest_name = src.name if not name else name
+        src.rename(dest_abs / dest_name)
+        return Path(dest_abs / dest_name)
+    return False
+
+
+def rm_tree(pth):
+    pth = Path(pth)
+    if not pth.exists(): return
+    for child in pth.glob('*'):
+        if child.is_file():
+            child.unlink()
+        else:
+            rm_tree(child)
+    pth.rmdir()
+
+
+def get_version_tag():
+    '''Returns the current git tag version.
+    '''
+    if Path('.git') is None or shutil.which('git') is None:
+        sys.exit(f"Unable to find `.git` directory or git is not installed.")
+
+    git_tag = subprocess.run(
+        ["git", "describe", "--tags", "--abbrev=0"],
```

Comment:
```
(armory)david.slater@noether:~/armory$ git describe
v0.15.4-83-g2906e99
(armory)david.slater@noether:~/armory$ armory --version
0.15.5.g2906e99
```

For tagging docker container, see armory version above.

Reply:
Fixed using `scm` from `setuptools`. Modified `pyproject.toml`, `version.py`, and `build.py` to all use `setuptools_scm` along with the same formatting arguments.

Developers are likely to experience version mismatches if a file is edited while/during the building of an image due to the versioning requirements, highly dynamic/volatile nature of the file system during development, and the interdependence of `armory` on `docker`(due, in part, to the [available configuration options](https://github.com/pypa/setuptools_scm/blob/c4d37004ab0a16c2cacdc58ef06b36956db02a9f/setup.cfg) within `scm` as well as `docker` tag syntax. ).